src/core/server/elasticsearch/cluster_client.test.ts

```diff
@@ -0,0 +1,373 @@
+
+    expect(mockParseElasticsearchClientConfig).toHaveBeenCalledTimes(1);
+    expect(mockParseElasticsearchClientConfig).toHaveBeenLastCalledWith(mockEsConfig, mockLogger, {
+      auth: false,
+      ignoreCertAndKey: false,
+    });
+  });
+
+  test('passes only filtered headers to the scoped cluster client', () => {
+    clusterClient.asScoped({ headers: { zero: '0', one: '1', two: '2', three: '3' } });
+
+    expect(MockScopedClusterClient).toHaveBeenCalledTimes(1);
+    expect(MockScopedClusterClient).toHaveBeenCalledWith(
+      expect.any(Function),
+      expect.any(Function),
+      { one: '1', two: '2' }
+    );
+  });
+
+  test('both scoped and internal API caller fail if cluster client is closed', async () => {
+    clusterClient.asScoped({ headers: { zero: '0', one: '1', two: '2', three: '3' } });
+
+    clusterClient.close();
+
+    const [[internalAPICaller, scopedAPICaller]] = MockScopedClusterClient.mock.calls;
+    await expect(internalAPICaller('ping')).rejects.toThrowErrorMatchingInlineSnapshot(
+      `"Cluster client cannot be used after it has been closed."`
+    );
+
+    await expect(scopedAPICaller('ping', {})).rejects.toThrowErrorMatchingInlineSnapshot(
+      `"Cluster client cannot be used after it has been closed."`
+    );
+  });
+});
+
+describe('#close', () => {
+  let mockEsClientInstance: { close: jest.Mock };
+  let mockScopedEsClientInstance: { close: jest.Mock };
+
+  let clusterClient: ClusterClient;
+
+  beforeEach(() => {
+    mockEsClientInstance = { close: jest.fn() };
+    mockScopedEsClientInstance = { close: jest.fn() };
+    MockClient.mockImplementationOnce(() => mockEsClientInstance).mockImplementationOnce(
+      () => mockScopedEsClientInstance
+    );
+
+    clusterClient = new ClusterClient(
+      { apiVersion: 'es-version', requestHeadersWhitelist: [] } as any,
+      logger.get()
+    );
+  });
+
+  test('closes underlying Elasticsearch client', () => {
+    expect(mockEsClientInstance.close).not.toHaveBeenCalled();
+
+    clusterClient.close();
+    expect(mockEsClientInstance.close).toHaveBeenCalledTimes(1);
+    mockEsClientInstance.close.mockClear();
+
+    clusterClient.close();
+    expect(mockEsClientInstance.close).not.toHaveBeenCalled();
+  });
+
+  test('closes both internal and scoped underlying Elasticsearch clients', () => {
+    clusterClient.asScoped({ headers: { one: '1' } });
+
+    expect(mockEsClientInstance.close).not.toHaveBeenCalled();
+    expect(mockScopedEsClientInstance.close).not.toHaveBeenCalled();
+
+    clusterClient.close();
+    expect(mockEsClientInstance.close).toHaveBeenCalledTimes(1);
+    expect(mockScopedEsClientInstance.close).toHaveBeenCalledTimes(1);
+    mockEsClientInstance.close.mockClear();
+    mockScopedEsClientInstance.close.mockClear();
+
+    clusterClient.close();
+    expect(mockEsClientInstance.close).not.toHaveBeenCalled();
+    expect(mockScopedEsClientInstance.close).not.toHaveBeenCalledTimes(1);
```

Comment:
1 **nit:** `.not.toHaveBeenCalled();` for consistency 
2 could be extracted in a separate test to emphasise an intention - `does not call close on already closed client`-ish

Reply:
yes and yes :+1: